core/shared/src/main/scala/scalaz/zio/Fiber.scala

```diff
@@ -81,6 +117,7 @@ object Fiber {
     new Fiber[E, A] {
       def join: IO[E, A]                                     = IO.point(a)
       def interrupt0(ts: List[Throwable]): IO[Nothing, Unit] = IO.unit
+      def finished(f: ExitResult[E, A] => IO[Nothing, Unit]) = IO.unit
```

Comment:
I don't think this is correct. I think it must be:

```scala
def finished(f: ExitResult[E, A] => IO[Nothing, Unit]) = f(ExitResult.Completed(a)).fork.void
```

We should have a guarantee that `finished` callbacks are _always_ called, whether a fiber starts in a done state or whether it's actually running to completion.

I decided to fork it because who knows how much time it will take (better to be safer).

Reply:
We should have a test for this, probably.